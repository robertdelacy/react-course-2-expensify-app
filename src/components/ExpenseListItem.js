import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {moment(createdAt).format('Do MMMM, YYYY')}</p>
    </div>
);

export const Test = () => (
    <div>
        <Link to={`/edit/10`}>
            <h3>Coffee</h3>
        </Link>
        <p>Â£3.10 12th February, 2019</p>
    </div>
);

export default ExpenseListItem;