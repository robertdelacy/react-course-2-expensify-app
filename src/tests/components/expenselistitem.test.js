import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import { expenses } from '../fixtures';

let wrapper;

beforeEach(() => {
    wrapper =  shallow(<ExpenseListItem {...expenses[0]}/>);
});

test('should render ExpenseListItem with expenses', () => {
    expect(wrapper).toMatchSnapshot();
});