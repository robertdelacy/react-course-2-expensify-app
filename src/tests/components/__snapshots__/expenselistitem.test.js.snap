// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <Link
    to="/edit/0"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    195
     - 
    1st January, 1970
  </p>
</div>
`;
